sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedProcessorTest.java

```diff
@@ -332,10 +315,136 @@ public void staledLeaseAcquiring() {
+            .options(new ChangeFeedProcessorOptions()
+                .setLeasePrefix("TEST")
+                .setStartFromBeginning(true)
+                .setMaxItemCount(10)
+            )
+            .build();
+
+        changeFeedProcessor.start().subscribeOn(Schedulers.elastic())
+            .timeout(Duration.ofMillis(2 * CHANGE_FEED_PROCESSOR_TIMEOUT))
+            .onErrorResume(throwable -> {
+                log.error("Change feed processor did not start in the expected time", throwable);
+                return Mono.error(throwable);
+            })
+            .doOnSuccess(aVoid -> {
+                // Wait for the feed processor to receive and process the first batch of documents.
+                waitToReceiveDocuments(2 * CHANGE_FEED_PROCESSOR_TIMEOUT, FEED_COUNT);
+            })
+            .then(
+                // increase throughput to force a single partition collection to go through a split
+                createdFeedCollectionForSplit.readProvisionedThroughput().subscribeOn(Schedulers.elastic())
+                    .flatMap(currentThroughput ->
+                        createdFeedCollectionForSplit.replaceProvisionedThroughput(FEED_COLLECTION_THROUGHPUT).subscribeOn(Schedulers.elastic())
+                    )
+                .then()
+            )
+            .subscribe();
+
+        // Wait for the feed processor to receive and process the first batch of documents and apply throughput change.
+        try {
+            Thread.sleep(4 * CHANGE_FEED_PROCESSOR_TIMEOUT);
+        } catch (InterruptedException e) {
+            log.error(e.getMessage());
+        }
+
+        // Loop through reading the current partition count until we get a split
+        //   This can take up to two minute or more.
+        String partitionKeyRangesPath = extractContainerSelfLink(createdFeedCollectionForSplit);
+        FeedOptions feedOptions = new FeedOptions();
+        feedOptions.setRequestContinuation(null);
+
+        AsyncDocumentClient contextClient = getContextClient(createdDatabase);
+        Flux.just(1).subscribeOn(Schedulers.elastic())
+            .flatMap( value -> {
+                log.warn("Reading current hroughput change.");
+                return contextClient.readPartitionKeyRanges(partitionKeyRangesPath, feedOptions);
+            })
+            .map(partitionKeyRangeFeedResponse -> {
+                int count = partitionKeyRangeFeedResponse.getResults().size();
+
+                if ( count < 2) {
+                    log.warn("Throughput change is pending.");
+                    throw new RuntimeException("Throughput change is not done.");
+                }
+                return count;
+            })
+            // this will timeout approximately after 3 minutes
+            .retry(40, throwable -> {
+                try {
+                    log.warn("Retrying...");
+                    Thread.sleep(CHANGE_FEED_PROCESSOR_TIMEOUT);
+                } catch (InterruptedException e) {
```

Comment:
don't catch `InterruptedException` if that's thrown, we can fail the test.

Reply:
here and elsewhere please. we don't need to catch this, let it get thrown

Reply:
The try/catch is needed to avoid compilation errors such as: "Error:(440, 29) java: unreported exception java.lang.InterruptedException; must be caught or declared to be thrown"

Reply:
makes sense here.
so please use this instead to translate checked to unchecked exception.
https://projectreactor.io/docs/core/release/api/reactor/core/Exceptions.html#propagate-java.lang.Throwable-

if `InterruptedException` we should error out.

Reply:
Good suggestion, but it goes way beyond what we need to address at this time which is to release a fix for the CFP ASAP.

Feel free to open an issue and assign it to me to refactor these tests and remove the handling of the exception.

Reply:
@milismsft I have already signed off on v3, and I agree with you on v3 as this is hotfix for v3 it should go out as soon as possible.

on v4 we should do the right thing (this is not in prod yet) rather than introducing technical debt.

Reply:
V4 latest release is currently used by couple customers a preview.

Reply:
I'll deal with the debt later; the changes you're requesting go beyond what this PR/work item should address. IMHO et's keep things simple enough in case we need to review this later and use a different PR to address  other things.